### PR TITLE
[fastlane] fix prompt crash on multi_line_end_keyword: "^D"

### LINKED ISSUE #14185: prompt crashes on multi_line_end_keyword: "^D"
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description
prompt crashes on `multi_line_end_keyword: "^D"`

##### Command executed
```ENV["NOTES"] = prompt(text: "Write some release notes:", multi_line_end_keyword: "^D")```

##### Complete output when running fastlane, including the stack trace and command used
<details>
  <pre>+------+---------------------------------------------------------------+-------------+
|                                  fastlane summary                                  |
+------+---------------------------------------------------------------+-------------+
| Step | Action                                                        | Time (in s) |
+------+---------------------------------------------------------------+-------------+
| 1    | Verifying fastlane version                                    | 0           |
| 2    | default_platform                                              | 0           |
| 3    | Switch to ios check_for_changes lane                          | 0           |
| 4    | prompt                                                        | 1           |
| 5    | Switch to ios set_build_number lane                           | 0           |
| 6    | git tag --merged HEAD | grep 'beta-[0-9][0-9]*\.' | sed 's/.* | 0           |
| 7    | date -u +%y%m%d%H%M                                           | 0           |
| 8    | /usr/libexec/PlistBuddy -c 'Set :CFBundleShortVersionString 1 | 0           |
| 9    | /usr/libexec/PlistBuddy -c 'Set :CFBundleVersion 1902051425'  | 0           |
| 💥   | prompt                                                        | 2           |
+------+---------------------------------------------------------------+-------------+

[15:25:08]: fastlane finished with errors

Looking for related GitHub issues on fastlane/fastlane...

➡️  NoMethodError: [!] undefined method `real_path' for nil:NilClass
    https://github.com/fastlane/fastlane/issues/13667 [closed] 10 💬
    3 weeks ago

➡️  [set_info_plist_value] add option to merge with map
    https://github.com/fastlane/fastlane/pull/10514 [open] 22 💬
    14 Nov 2018

➡️  Each warning log is duplicated output 80-120x
    https://github.com/fastlane/fastlane/issues/13470 [closed] 4 💬
    4 weeks ago

and 18 more at: https://github.com/fastlane/fastlane/search?q=undefined%20method%20%60index%27%20for%20nil&type=Issues&utf8=✓

🔗  You can ⌘ + double-click on links to open them directly in your browser.
bundler: failed to load command: fastlane (/Users/slider/.rbenv/versions/2.3.1/bin/fastlane)
NoMethodError: [!] undefined method `index' for nil:NilClass
  /Users/slider/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/fastlane-2.115.0/fastlane/lib/fastlane/actions/prompt.rb:21:in `block in run'
  /Users/slider/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/fastlane-2.115.0/fastlane/lib/fastlane/actions/prompt.rb:19:in `loop'
  /Users/slider/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/fastlane-2.115.0/fastlane/lib/fastlane/actions/prompt.rb:19:in `run'
  /Users/slider/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/fastlane-2.115.0/fastlane/lib/fastlane/runner.rb:253:in `block (2 levels) in execute_action'
  /Users/slider/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/fastlane-2.115.0/fastlane/lib/fastlane/actions/actions_helper.rb:50:in `execute_action'
  /Users/slider/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/fastlane-2.115.0/fastlane/lib/fastlane/runner.rb:232:in `block in execute_action'
  /Users/slider/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/fastlane-2.115.0/fastlane/lib/fastlane/runner.rb:227:in `chdir'
  /Users/slider/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/fastlane-2.115.0/fastlane/lib/fastlane/runner.rb:227:in `execute_action'
  /Users/slider/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/fastlane-2.115.0/fastlane/lib/fastlane/runner.rb:157:in `trigger_action_by_name'
  /Users/slider/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/fastlane-2.115.0/fastlane/lib/fastlane/fast_file.rb:154:in `method_missing'
  Fastfile:95:in `block (2 levels) in parsing_binding'
  /Users/slider/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/fastlane-2.115.0/fastlane/lib/fastlane/lane.rb:33:in `call'
  /Users/slider/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/fastlane-2.115.0/fastlane/lib/fastlane/runner.rb:49:in `block in execute'
  /Users/slider/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/fastlane-2.115.0/fastlane/lib/fastlane/runner.rb:45:in `chdir'
  /Users/slider/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/fastlane-2.115.0/fastlane/lib/fastlane/runner.rb:45:in `execute'
  /Users/slider/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/fastlane-2.115.0/fastlane/lib/fastlane/lane_manager.rb:56:in `cruise_lane'
  /Users/slider/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/fastlane-2.115.0/fastlane/lib/fastlane/command_line_handler.rb:36:in `handle'
  /Users/slider/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/fastlane-2.115.0/fastlane/lib/fastlane/commands_generator.rb:108:in `block (2 levels) in run'
  /Users/slider/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:178:in `call'
  /Users/slider/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:153:in `run'
  /Users/slider/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/commander-fastlane-4.4.6/lib/commander/runner.rb:476:in `run_active_command'
  /Users/slider/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/fastlane-2.115.0/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:76:in `run!'
  /Users/slider/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/commander-fastlane-4.4.6/lib/commander/delegates.rb:15:in `run!'
  /Users/slider/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/fastlane-2.115.0/fastlane/lib/fastlane/commands_generator.rb:333:in `run'
  /Users/slider/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/fastlane-2.115.0/fastlane/lib/fastlane/commands_generator.rb:41:in `start'
  /Users/slider/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/fastlane-2.115.0/fastlane/lib/fastlane/cli_tools_distributor.rb:113:in `take_off'
  /Users/slider/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/fastlane-2.115.0/bin/fastlane:23:in `<top (required)>'
  /Users/slider/.rbenv/versions/2.3.1/bin/fastlane:23:in `load'
  /Users/slider/.rbenv/versions/2.3.1/bin/fastlane:23:in `<top (required)>'
 </pre>
</details>

### Environment

<details>
  <pre> <details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                       |
| --------------------------- | ------------------------------------------- |
| OS                          | 10.14.3                                     |
| Ruby                        | 2.3.1                                       |
| Bundler?                    | true                                        |
| Git                         | git version 2.17.2 (Apple Git-113)          |
| Installation Source         | ~/.rbenv/versions/2.3.1/bin/fastlane        |
| Host                        | Mac OS X 10.14.3 (18D42)                    |
| Ruby Lib Dir                | ~/.rbenv/versions/2.3.1/lib                 |
| OpenSSL Version             | OpenSSL 1.0.2j  26 Sep 2016                 |
| Is contained                | false                                       |
| Is homebrew                 | false                                       |
| Is installed via Fabric.app | false                                       |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version               | 10.1                                        |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   | en_US.UTF-8 | ✅ |
| LANGUAGE |             |   |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
# Customise this file, documentation can be found here:
# https://github.com/fastlane/fastlane/tree/master/fastlane/docs
# All available actions: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Actions.md
# can also be listed using the `fastlane actions` command

# Change the syntax highlighting to Ruby
# All lines starting with a # are ignored when running `fastlane`

# If you want to automatically update fastlane if a new version is available:
# update_fastlane

# This is the minimum version number required.
# Update this, if you use features of a newer version
fastlane_version "1.109.0"

default_platform :ios

project = "pos"
plist = "#{project}/Info.plist"

ENV["CRASHLYTICS_API_TOKEN"] = "04fcf32f8ff712d0aa3048efbb82ec027ab7d670"
ENV["CRASHLYTICS_BUILD_SECRET"] = "d2e6566bfc836af2f81f34a4ce21f1513118211787aadf198300638c3067fe02"
ENV["FL_UPLOAD_SYMBOLS_TO_CRASHLYTICS_DSYM_PATH"] = "./#{project}.app.dSYM.zip"
ENV["CRASHLYTICS_GROUPS"] = "internal"
ENV["TESTFLIGHT_GROUPS"] = "Shoptiques"

platform :ios do
  before_all do
    ENV["VERSION"] = ""
    ENV["TAG"] = ""
    ENV["NOTES"] = ""
    ENV["LANE_NAME"] = lane_context[SharedValues::LANE_NAME].gsub('ios ', '')
  end

  # desc "Runs all the tests"
  # lane :test do
  #   scan
  # end

  desc "Update debug profiles & certs"
  lane :updatedebug do
    sigh(output_path: "./fastlane/profiles", development: true)
  end

  desc "Update beta profiles & certs"
  lane :updatebeta do
    create_certs(method: "ad-hoc")
  end

  desc "Update release profiles & certs"
  lane :updaterelease do
    create_certs(method: "app-store")
  end

  desc "Submit a new Build to RainforestQA"
  lane :rainprod do
    check_for_changes
    xcodebuild(
      scheme: project,
      workspace: "#{project}.xcworkspace",
      configuration: "Release",
      sdk: "iphonesimulator",
      xcargs: "SYMROOT='/var/tmp'",
    )
    zip(
      path: "/var/tmp/Release-iphonesimulator/#{project}.app",
      output_path: "#{project}.app.zip",
    )
    sh("aws s3 cp ../#{project}.app.zip s3://shoptiques-shop/ios/ --acl public-read --quiet")
    sh("open https://app.rainforestqa.com/run_groups/2667")
  end

  desc "Submit a new Build to RainforestQA"
  lane :rainbeta do
    check_for_changes
    xcodebuild(
      scheme: project,
      workspace: "#{project}.xcworkspace",
      configuration: "Debug",
      sdk: "iphonesimulator",
      xcargs: "SYMROOT='/var/tmp'",
    )
    zip(
      path: "/var/tmp/Debug-iphonesimulator/#{project}.app",
      output_path: "#{project}.app.zip",
    )
    sh("aws s3 cp ../#{project}.app.zip s3://shoptiques-shop/ios/#{project}-dev.app.zip --acl public-read --quiet")
    sh("open https://app.rainforestqa.com/run_groups/2899")
  end

  desc "Submit a new Beta Build to Crashlytics"
  lane :beta do
    check_for_changes
    set_build_number
    ENV["NOTES"] = prompt(text: "Write some release notes:", multi_line_end_keyword: "^D")
    build(method: "ad-hoc")
    send_to_crashlytics
    git_create_tag
  end

  desc "Deploy a new version to the App Store & distribute via TestFlight"
  lane :prod do
    check_for_changes
    set_build_number
    ENV["NOTES"] = prompt(text: "Write some release notes:", multi_line_end_keyword: "^D")
    build(method: "app-store")
    testflight(
      changelog: ENV["NOTES"],
      distribute_external: true,
      groups: ENV["TESTFLIGHT_GROUPS"],
    )
    upload_symbols_to_crashlytics
    git_create_tag
  end

  desc "Deploy a new version to the App Store & distribute via TestFlight & Crashlytics"
  lane :betaprod do
    check_for_changes
    set_build_number
    ENV["NOTES"] = prompt(text: "Write some release notes:", multi_line_end_keyword: "^D")
    build(method: "ad-hoc")
    send_to_crashlytics
    build(method: "app-store")
    testflight(
      changelog: ENV["NOTES"],
      distribute_external: true,
      groups: ENV["TESTFLIGHT_GROUPS"],
    )
    upload_symbols_to_crashlytics
    git_create_tag
  end

  desc "Deploy an upcoming version to TestFlight"
  lane :pre do
    check_for_changes
    set_build_number
    build(method: "app-store")
    testflight(
      changelog: "test",
      distribute_external: true,
      groups: "Rok",
    )
    upload_symbols_to_crashlytics
    git_create_tag
  end

  desc "Deploy a new version to the App Store"
  lane :release do
    check_for_changes
    set_build_number
    build(method: "app-store")
    deliver
    upload_symbols_to_crashlytics
    git_create_tag
  end

  ######################### PRIVATE LANES #########################

  private_lane :create_certs do |options|
    cert(output_path: "./fastlane/certificates")
    sigh(output_path: "./fastlane/profiles", adhoc: options[:method] == "ad-hoc")
  end

  private_lane :set_build_number do
    lane = ENV["LANE_NAME"]
    tag = sh("git tag --merged HEAD | grep '#{lane}-[0-9][0-9]*\\.' | sed 's/.*-\\([0-9.]*\\)-.*/\\1/g' | sort -V | tail -n1").gsub("\n", '').split(".")
    tag = sh("git tag --merged HEAD | grep '[0-9][0-9]*\\.' | sed 's/.*-\\([0-9.]*\\)-.*/\\1/g' | sort -V | tail -n1").gsub("\n", '').split(".") if tag.empty?
    major = tag[0] || 0
    minor = tag[1] || 0
    fix = tag[2] || 0
    oldVer = (fix == 0 ? [major,minor] : [major,minor,fix]).join(".")
    puts("New version (enter to keep old version):")
    newVer = STDIN.gets.chomp.strip
    version = newVer.empty? ? oldVer : newVer
    build = sh("date -u +%y%m%d%H%M").gsub("\n", '')

    # using PlistBuddy instead of set_info_plist_value (CircleCI has some problems with it)
    sh("/usr/libexec/PlistBuddy -c 'Set :CFBundleShortVersionString #{version}' ../#{plist}")
    sh("/usr/libexec/PlistBuddy -c 'Set :CFBundleVersion #{build}' ../#{plist}")
    ENV["VERSION"] = "v#{version} (#{build})"
    ENV["TAG"] = ENV["LANE_NAME"] + "-#{version}-#{build}"
  end

  private_lane :check_for_changes do |options|
    unless Helper.backticks("git status | grep modified").empty? || prompt(text: "Do you want to continue?", boolean: true)
      UI.user_error!("Unwanted modified files found.")
    end
  end

  private_lane :build do |options|
    cocoapods
    gym(
      scheme: project,
      configuration: options[:method] == "app-store" ? "Release" : "Beta",
      export_method: options[:method],
      include_bitcode: false,
    )
  end

  private_lane :send_to_crashlytics do
    crashlytics(notes: ENV["NOTES"]) || crashlytics(notes: ENV["NOTES"]) || crashlytics(notes: ENV["NOTES"]) || crashlytics(notes: ENV["NOTES"]) || crashlytics(notes: ENV["NOTES"]) || crashlytics(notes: ENV["NOTES"]) || crashlytics(notes: ENV["NOTES"]) || crashlytics(notes: ENV["NOTES"]) || crashlytics(notes: ENV["NOTES"]) || crashlytics(notes: ENV["NOTES"]) || crashlytics(notes: ENV["NOTES"]) || crashlytics(notes: ENV["NOTES"]) || crashlytics(notes: ENV["NOTES"]) || crashlytics(notes: ENV["NOTES"])
  end

  private_lane :git_create_tag do
    add_git_tag(tag: ENV["TAG"])
    push_git_tags(remote: "origin", tag: ENV["TAG"])
  end

  after_all do |lane|
    puts("version: " + ENV["VERSION"])
    puts("tag: " + ENV["TAG"])
    puts("notes: " + ENV["NOTES"])
  end

  error do |lane, exception|
    puts("version: " + ENV["VERSION"])
    puts("tag: " + ENV["TAG"])
    puts("notes: " + ENV["NOTES"])
  end
end


# More information about multiple platforms in fastlane: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Platforms.md
# All available actions: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Actions.md

# fastlane reports which actions are used
# No personal data is recorded. Learn more at https://github.com/fastlane/enhancer

```
</details>

<details><summary>`./fastlane/Appfile`</summary>

```ruby
apple_id ENV["APPLE_ID"] || "marc@shoptiques.com" # Your Apple email address

team_id "GYT7FDWCLH"  # Developer Portal Team ID
itc_team_id 1523967

for_lane :updatedebug do
	app_identifier "com.shoptiques.pos.debug"
end

for_lane :updatebeta do
	app_identifier "com.shoptiques.pos.beta"
end

for_lane :beta do
	app_identifier "com.shoptiques.pos.beta"
end

for_lane :updaterelease do
	app_identifier "com.shoptiques.pos"
end

for_lane :release do
	app_identifier "com.shoptiques.pos"
end

# More information: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Appfile.md

```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.115.0 | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| did_you_mean              | 1.0.0        |
| bundler                   | 1.16.2       |
| io-console                | 0.4.6        |
| CFPropertyList            | 3.0.0        |
| concurrent-ruby           | 1.0.5        |
| i18n                      | 0.9.5        |
| minitest                  | 5.11.3       |
| thread_safe               | 0.3.6        |
| tzinfo                    | 1.2.5        |
| activesupport             | 4.2.10       |
| public_suffix             | 2.0.5        |
| addressable               | 2.6.0        |
| atomos                    | 0.1.3        |
| babosa                    | 1.0.2        |
| claide                    | 1.0.2        |
| fuzzy_match               | 2.0.4        |
| nap                       | 1.1.0        |
| cocoapods-core            | 1.5.3        |
| cocoapods-deintegrate     | 1.0.2        |
| cocoapods-downloader      | 1.2.1        |
| cocoapods-plugins         | 1.0.0        |
| cocoapods-search          | 1.0.0        |
| cocoapods-stats           | 1.0.0        |
| netrc                     | 0.11.0       |
| cocoapods-trunk           | 1.3.1        |
| cocoapods-try             | 1.1.0        |
| colored2                  | 3.1.2        |
| escape                    | 0.0.4        |
| fourflusher               | 2.0.1        |
| gh_inspector              | 1.1.3        |
| molinillo                 | 0.6.6        |
| ruby-macho                | 1.3.1        |
| nanaimo                   | 0.2.6        |
| xcodeproj                 | 1.8.0        |
| cocoapods                 | 1.5.3        |
| colored                   | 1.2          |
| highline                  | 1.7.10       |
| commander-fastlane        | 4.4.6        |
| declarative               | 0.0.10       |
| declarative-option        | 0.1.0        |
| digest-crc                | 0.4.1        |
| unf_ext                   | 0.0.7.5      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20180417 |
| dotenv                    | 2.6.0        |
| emoji_regex               | 1.0.1        |
| excon                     | 0.62.0       |
| multipart-post            | 2.0.0        |
| faraday                   | 0.15.4       |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| faraday_middleware        | 0.13.1       |
| fastimage                 | 2.1.5        |
| jwt                       | 2.1.0        |
| memoist                   | 0.16.0       |
| multi_json                | 1.13.1       |
| os                        | 1.0.0        |
| signet                    | 0.11.0       |
| googleauth                | 0.6.7        |
| httpclient                | 2.8.3        |
| mime-types-data           | 3.2018.0812  |
| mime-types                | 3.2.2        |
| uber                      | 0.1.0        |
| representable             | 3.0.4        |
| retriable                 | 3.1.2        |
| google-api-client         | 0.23.9       |
| google-cloud-env          | 1.0.5        |
| google-cloud-core         | 1.3.0        |
| google-cloud-storage      | 1.16.0       |
| json                      | 2.1.0        |
| mini_magick               | 4.5.1        |
| multi_xml                 | 0.6.0        |
| plist                     | 3.5.0        |
| rubyzip                   | 1.2.2        |
| security                  | 0.1.3        |
| naturally                 | 2.2.0        |
| simctl                    | 1.6.5        |
| slack-notifier            | 2.3.2        |
| terminal-notifier         | 1.8.0        |
| unicode-display_width     | 1.4.1        |
| terminal-table            | 1.8.0        |
| tty-screen                | 0.6.5        |
| tty-cursor                | 0.6.0        |
| tty-spinner               | 0.9.0        |
| word_wrap                 | 1.0.0        |
| rouge                     | 2.0.7        |
| xcpretty                  | 0.3.0        |
| xcpretty-travis-formatter | 1.0.0        |
</details>

*generated on:* **2019-02-05**
</details> </pre>
</details>

### PR DESCRIPTION
**Pull request by mollyIV**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context

`prompt` action crashes when `multi_line_end_keyword` detects end of file character equals to `ctrl-d` (when you press Ctrl-d while inputting data).

closes https://github.com/fastlane/fastlane/issues/14185

### Description

According to [Ruby docs](http://ruby-doc.org/core-2.1.0/IO.html#method-i-gets) `STDIN.gets`:

> Returns nil if called at end of file

Basically when you hit `ctrl-d` while inserting the lines of text to the prompt, you received:

```bash
.../fastlane/fastlane/lib/fastlane/actions/prompt.rb:22:in `block in run': [!] undefined method `index' for nil:NilClass (NoMethodError)
```
The fix is about checking whether `STDIN.gets` return a nil value which represents the end of file symbol 🎉 

I tested the changes by inserting two lines of text and hitting `ctrl-d`. 
I was trying to add a unit test but couldn't figure out how to mock returning `nil` value from `STDIN.gets` 😞 If you have any ideas how we could do that I am all ears 🙏 

Cheers 🙇

### CODE REVIEW

**Review by joshdholtz — APPROVED** [review]
This is great! I actually never thought about using `^D` before but I think I might start doing this 😊
